include/RAJA/index/ListSegment.hpp

```diff
@@ -84,11 +88,83 @@ namespace RAJA
 template <typename T>
 class TypedListSegment
 {
+
+#if defined(RAJA_ENABLE_CUDA)
+  static constexpr bool Has_CUDA = true;
+#else
+  static constexpr bool Has_CUDA = false;
+#endif
+
+  //! tag for trivial per-element copy
+  struct TrivialCopy {
+  };
+  //! tag for memcpy-style copy
+  struct BlockCopy {
+  };
+
+  //! alias for GPU memory tag
+  using GPU_memory = std::integral_constant<bool, true>;
```

Comment:
I like all of this, am wondering if it would be worth factoring this into RAJAVec in a subsequent PR so we can leverage it elsewhere as well.  StaticIndexSet for example.